### ISSUE TITLE
0.10.2 failing

### ISSUE THREAD

**Issue by homonto** [pstolarz/OneWireNg#34]
Hi,

So after installing 0.10.2 over 0.10.1 I started having totally inconsistent reading.
I have 3 DS18B20 and so far all was OK.
The moment I upgrade the program has issues with finding sensors, or, invalid reading 
Actually I am not able to say which one is when.
As per your example, I get either:
`Invalid CRC!`
or simply the sensor is omitted

Reverting to 0.10.1 solves the problem.
I did not change anything in my code, so maybe 0.10.2 has some breaking changes?

**Comment by pstolarz** [comment]
What is your platform?
Change:
```c
# define CONFIG_BITBANG_TIMING TIMING_RELAXED
```
to
```c
# define CONFIG_BITBANG_TIMING TIMING_STRICT
```
in file `src/OneWireNg_Config.h` and let me know about results.

**Comment by homonto** [comment]
it works with this setting
with relaxed it is... too relaxed ;-)
My system: ESP32-S2, IDE: Arduino, 3 sensors DS18B20 with 3 wires and 1 resistor connected to GPIO10 on ESP.

Let me know if you need more info.
Since these 3 sensors were measuring the temperature of my Power Supply, since today morning my Power Supply is not working (over/under temperature protection is based on them) - I need to find a way how to handle false alarms like this ;-)

**Comment by pstolarz** [comment]
I tested the relaxed timings on my ESP32 devkit (legacy version) and worked fine. I have some S2 board and will check the relaxed timing soon. I'm suspecting you use WiFi communication extensively on your setup with 1-wire sensors read-outs and this causes the problem you observed (WiFi interrupts re-entrant time critical 1-wire bit-bangs).

I will change the timing to strict, thanks for the catch.

> Since these 3 sensors were measuring the temperature of my Power Supply, since today morning my Power Supply is not working (over/under temperature protection is based on them) - I need to find a way how to handle false alarms like this ;-)

Do you mean I spoiled something? ;)

**Comment by homonto** [comment]
;-) don't worry - this Linear Power Supply is 5V 10A for my Raspberry Pi farm. I upgraded libs today and it automatically went to 0V due to over temperature. It took me some time to diagnose, especially that Arduino is caching the compiled libs so simple downgrade of library was not successful - I had to ... you know - this and that. But finally I narrowed the issue to be the upgrade so all is back normal - _"u mnie TERAZ dziala"_ as admin would say ;-)

**Comment by homonto** [comment]
With regards to WiFi - the power supply provides data to Home Assistant over MQTT so yes, WiFi always ON and MQTT on top of that. I could try to turn off WiFi fo the time when library is measuring - I will try. But it will then take time to reestablish connectivity - I will test and report sir!

**Comment by pstolarz** [comment]
Dzięki i fajnie, że rodacy też używają mojej biblioteki. Pozdrawiam ;)

**Comment by homonto** [comment]
> Dzięki i fajnie, że rodacy też używają mojej biblioteki. Pozdrawiam ;)

I usually use SHT31 for my sensors, but since I needed more of them I decided to use DS18B20. I started with the standard onewire - NOT WORKING AT ALL! Yours is superb - before upgrade I mean ;-)

**Comment by homonto** [comment]
Now working again!
Thx.

<img width="402" alt="Screenshot 2022-03-08 at 18 10 36" src="https://user-images.githubusercontent.com/46562447/157299650-8dcab2fe-d06d-402b-ac97-47efdcf21734.png">
<img width="408" alt="Screenshot 2022-03-08 at 18 10 52" src="https://user-images.githubusercontent.com/46562447/157299664-0ec9cd9b-09a9-49d3-a8a4-d151ea8e9eaf.png">

**Comment by pstolarz** [comment]
This is with latest master patch, right?
BTW: What is this soft for controlling the power-supply you use?

**Comment by homonto** [comment]
I did not apply the patch yet - I understood the only change is what you already suggested, right?
The entire software is my "wypociny" with ESP32.
The screenshot is from Home Assistant.

**Comment by pstolarz** [comment]
Nie bądź taki skromny :-)

**Comment by homonto** [comment]
jak mi dasz maila to pogadamy przeciez bo widze, ze rodak z Krakowa ;)

**Comment by pstolarz** [comment]
pstolarz@o2.pl

**Comment by uzi18** [comment]
@pstolarz  Pewnie że używają :) 

Waiting for platform.io update for this lib. 
Got strange behavior on esp32 - method begin() from dallastemperature scans bus and found device. 
But... Fetched temp reports device as disconnected.
Need more investigation here.

**Comment by uzi18** [comment]
@homonto lepiej kanal na gitter.im zalozyc

**Comment by pstolarz** [comment]
@uzi18 Check the fix with strict timings and let me know about result. Looks like moving to relaxed timing in 0.10.2 was not clever move (I checked it on my ESP32 and worked fine but didn't launch extensive tests with WiFi). If you confirm the latest commit fixes the issue I'll release 0.10.3 with this fix immediately.

**Comment by pstolarz** [comment]
> @homonto lepiej kanal na gitter.im zalozyc
Dobry pomysł. Można tam pogadać nieklawiaturowo też?

**Comment by pstolarz** [comment]
Założyłem grupę OneWireNg na tym gitterze. Jakby co to zapraszam. Później dodam jeszcze info o tej grupie na README. Dzięki za pomysł.

**Comment by uzi18** [comment]
dodaj takie cos: 
[![Join the chat at https://gitter.im/OneWireNg/community](https://badges.gitter.im/OneWireNg/community.svg)](https://gitter.im/OneWireNg/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)

**Comment by pstolarz** [comment]
@homonto @uzi18 I've just released a new 0.10.3 version of the lib with bit-banging timings set to strict by default. It will take some time to sync the lib with the lib-repos (PlatformIO, Arduino). I'd be grateful if you could check the new version on your ESP32 setups with WiFi communications and let me know about its stability now. Thanks for the catch!

**Comment by homonto** [comment]
you mean: shut down all my RPi? ;-)
I will check and will let you know
